src/app/tasks/components/tasks.component.ts

```diff
@@ -0,0 +1,141 @@
+import { Component, OnInit } from '@angular/core';
+import { TaskElement, TaskService } from '../service/task.service';
+import { Observable, Subscription } from 'rxjs';
+import { MatCheckboxChange } from '@angular/material/checkbox';
+import { select, Store } from '@ngrx/store';
+import { errorSelector, isLoadingSelector, taskSelector } from '../store/selectors';
+import { AppStateInterface } from '../types/appState.interface';
+
+@Component({
+  selector: 'app-tasks',
+  templateUrl: './tasks.component.html',
+  styleUrls: ['./tasks.component.css']
+})
+
+export class TasksComponent implements OnInit {
+  title = 'to-do-list-app';
+  isHiddenAddForm = true;
+  isHiddenEditForm = true;
+  listOfTasks: TaskElement[] = [];
+  listOfDoneTasks: boolean[] = [];
+  progress = 0;
+  newTaskValue = '';
+  newTask: TaskElement = {
+    id: '',
+    title: '',
+    status: false,
+  };
+  taskEdit: TaskElement = {
+    id: '',
+    title: '',
+    status: false
+  };
+  tasksArrayDB: any;
+  tasksArray: any;
+
+  isLoading$: Observable<boolean>;
+  error$: Observable<string | null>;
+  tasks$: Observable<TaskElement[]>
+
+  private subGetTasks$!: Subscription;
+
+  constructor(private taskService: TaskService, private store: Store<AppStateInterface>) {
+    this.isLoading$ = this.store.pipe(select(isLoadingSelector))
+    this.error$ = this.store.pipe(select(errorSelector))
+    this.tasks$ = this.store.pipe(select(taskSelector))
+  }
+
+  ngOnInit() {
+    // this.store.dispatch(TasksActions.getTasks());
+    this.getTasks();
+    for(let i=0; i<this.listOfTasks.length; i++){
+      this.listOfDoneTasks[i] = false;
+    }
+  }
+
+  getTasks(){
+    this.subGetTasks$ = this.taskService.getTasks().subscribe((tasksDB: Object) => {
```

Comment:
zmienić Object na inny typ, nie używa się go w taki sposób :) tak do codziennego użycia to przydaje się np.  Object.assign albo Object.keys